File: part-02/src/main/java/eu/luminis/workshop/smallsteps/logic/domainservice/catalog/LegoPartCatalog.java
package eu.luminis.workshop.smallsteps.logic.domainservice.catalog;

import eu.luminis.workshop.smallsteps.logic.domainmodel.valueobjects.LegoSetNumber;

import java.util.Map;

public interface LegoPartCatalog {
    Map<String, Integer> allPartsForLegoSet(LegoSetNumber legoSetNumber);
}

File: part-02/src/main/kotlin/eu/luminis/workshop/smallsteps/logic/domainService/catalog/LegoPartCatalog.kt
package eu.luminis.workshop.smallsteps.logic.domainService.catalog

import eu.luminis.workshop.smallsteps.logic.domainModel.valueObjects.LegoSetNumber

interface LegoPartCatalog {
    fun allPartsForLegoSet(legoSet: LegoSetNumber): Map<String, Int>?
}